full-stack-practice/client/src/lib/read.ts

```diff
@@ -0,0 +1,9 @@
+import { products, type Product } from './data';
+
+export function readCatalog(): Promise<Product[]> {
```

Comment:
You could have just changed the implementation in these files and left all the other code the same.